runtime/jcl/common/mgmtinit.c

```diff
@@ -953,8 +953,30 @@ getSegmentSizes(J9JavaVM *javaVM, J9MemorySegmentList *segList, U_64 *storedSize
 	omrthread_monitor_enter(segList->segmentMutex);
 
 	MEMORY_SEGMENT_LIST_DO(segList, seg)
+	if (isCodeCacheSegment) {
+		UDATA *mccCodeCache;
+		UDATA *warmAlloc;
+		UDATA *coldAlloc;
+		/* The JIT code cache grows from both ends of the segment: the warmAlloc pointer upwards from the start of the segment
+		 * and the coldAlloc pointer downwards from the end of the segment. The free space in a JIT code cache segment is the
+		 * space between the warmAlloc and coldAlloc pointers. See jit/codert/MultiCodeCache.hpp, the contract with the JVM is
+		 * that the address of the TR_MCCCodeCache structure is stored at the beginning of the segment, and that the warmAlloc
+		 * and coldAlloc pointers are the first two fields in TR_MCCCodeCache.
```

Comment:
The files and structure names are out of date.  The file that contains the base of the extensible `TR::CodeCache` object is `compiler/runtime/OMRCodeCache.hpp` in the Eclipse OMR project.  The concrete data structure is now referred to as `TR::CodeCache` (not `TR_MCCCodeCache`).